pyomo/core/expr/numeric_expr.py

```diff
@@ -1562,6 +1607,12 @@ def _decompose_linear_terms(expr, multiplier=1):
                 yield term
         else:
             raise LinearDecompositionError("Quadratic terms exist in a product expression.")
+    elif expr.__class__ is DivisionExpression:
+        if expr._args_[1].__class__ in native_numeric_types or not expr._args_[1].is_potentially_variable():
+            for term in _decompose_linear_terms(expr._args_[0], multiplier/expr._args_[1]):
```

Comment:
I assume that this iterator throws an exception if the argument passed is, in fact, nonlinear?

Reply:
Yes. that appears to be correct.